cmd/aries-agentd/startcmd/start.go

```diff
@@ -0,0 +1,166 @@
+
+// AgentInboundHostFlagShorthand is the flag shorthand for the agent inbound host command line argument.
+const AgentInboundHostFlagShorthand = "i"
+
+// AgentInboundHostFlagUsage is the usage text for the agent inbound host command line argument.
+const AgentInboundHostFlagUsage = "Inbound Host Name:Port"
+
+// AgentDBPathFlagName is the flag name for the database path command line argument.
+const AgentDBPathFlagName = "db-path"
+
+// AgentDBPathFlagShorthand is the flag shorthand name for the database path command line argument.
+const AgentDBPathFlagShorthand = "d"
+
+// AgentDBPathFlagUsage is the flag usage text for the database path command line argument.
+const AgentDBPathFlagUsage = "Path to database"
+
+// MissingHostErrorMessage is the error message to be shown when the user provides a blank host argument.
+const MissingHostErrorMessage = "Unable to start aries agentd, host not provided"
+
+// MissingInboundHostErrorMessage is the error message to be shown when the user provides a blank inbound host argument.
+const MissingInboundHostErrorMessage = "Unable to start aries agentd, HTTP Inbound transport host not provided"
+
+var logger = log.New("aries-framework/agentd")
+
+type server interface {
+	ListenAndServe(host string, router *mux.Router) error
+}
+
+// HTTPServer represents an actual server implementation.
+type HTTPServer struct{}
+
+// ListenAndServe starts the server using the standard Go HTTP server implementation.
+func (s *HTTPServer) ListenAndServe(host string, router *mux.Router) error {
+	return http.ListenAndServe(host, router)
+}
+
+// Cmd returns the Cobra start command.
+func Cmd(server server) *cobra.Command {
+	startCmd := &cobra.Command{
+		Use:   "start",
+		Short: "Start an agent",
+		Long:  `Start an Aries agent controller`,
+
+		Run: func(cmd *cobra.Command, args []string) {
+			host, err := cmd.Flags().GetString(AgentHostFlagName)
+			if err != nil {
+				logger.Errorf(err.Error())
+			}
+
+			inboundHost, err := cmd.Flags().GetString(AgentInboundHostFlagName)
+			if err != nil {
+				logger.Errorf(err.Error())
+			}
+
+			dbPath, err := cmd.Flags().GetString(AgentDBPathFlagName)
+			if err != nil {
+				logger.Errorf(err.Error())
+			}
+
+			err = startAgent(server, host, inboundHost, dbPath)
+			if err != nil {
+				logger.Errorf(err.Error())
+			}
+		},
+	}
+	startCmd.Flags().StringP(AgentHostFlagName, AgentHostFlagShorthand, "", AgentHostFlagUsage)
+	err := startCmd.MarkFlagRequired(AgentHostFlagName)
+	if err != nil {
+		logger.Errorf(err.Error())
+	}
+
+	startCmd.Flags().StringP(AgentInboundHostFlagName, AgentInboundHostFlagShorthand, "", AgentInboundHostFlagUsage)
+	err = startCmd.MarkFlagRequired(AgentInboundHostFlagName)
+	if err != nil {
+		logger.Errorf(err.Error())
+	}
+	startCmd.Flags().StringP(AgentDBPathFlagName, AgentDBPathFlagShorthand, "", AgentDBPathFlagUsage)
+	err = startCmd.MarkFlagRequired(AgentDBPathFlagName)
+	if err != nil {
+		logger.Errorf(err.Error())
+	}
+
```

Comment:
The "MarkFlagRequired" method throws an error if the flag name passed into it doesn't exist in the command. But it always will, so I'm not sure how I can actually test that. The linter will complain if I don't handle the error.